PROTO_tests/tests/io_test.py

```diff
@@ -1303,18 +1303,44 @@ def test_multi_index(self, dtype1, dtype2, size):
 
             assert isinstance(rd_idx, ak.MultiIndex)
             assert idx.to_list() == rd_idx.to_list()
+
         # handle categorical cases as well
-        if dtype1 == ak.str_:
-            t1 = ak.Categorical(t1)
-        if dtype2 == ak.str_:
-            t2 = ak.Categorical(t2)
-        idx = ak.Index.factory([t1, t2])
+        if ak.str_ in [dtype1, dtype2]:
+            if dtype1 == ak.str_:
+                t1 = ak.Categorical(t1)
+            if dtype2 == ak.str_:
+                t2 = ak.Categorical(t2)
+            idx = ak.Index.factory([t1, t2])
+            with tempfile.TemporaryDirectory(dir=TestHDF5.hdf_test_base_tmp) as tmp_dirname:
+                idx.to_hdf(f"{tmp_dirname}/idx_test")
+                rd_idx = ak.read_hdf(f"{tmp_dirname}/idx_test*")
+
+                assert isinstance(rd_idx, ak.MultiIndex)
+                assert idx.to_list() == rd_idx.to_list()
+
+    def test_hdf_overwrite_index(self):
+        # test repack with a single object
+        a = ak.Index(ak.arange(1000))
+        b = ak.Index(ak.randint(0, 100, 1000))
+        c = ak.Index(ak.arange(15))
```

Comment:
Is this worth parameterizing?

Reply:
It's not a for loop, so there's no way to parameterize. We use all three of the indices

Reply:
Makes sense.